zone/combat_record.h

```diff
@@ -0,0 +1,22 @@
+#ifndef EQEMU_COMBAT_RECORD_H
+#define EQEMU_COMBAT_RECORD_H
+
+#include <ctime>
+#include <string>
+
+class CombatRecord {
+public:
+	void Start(std::string in_mob_name);
+	void Stop();
+	bool InCombat();
+	void ProcessHPEvent(int hp, int current_hp);
+	double TimeInCombat() const;
+private:
+	std::string mob_name;
+	time_t      start_time      = 0;
```

Comment:
I would also think about std::chrono instead of old C time API, but this works so whatever

Reply:
`time_t` is incredibly lightweight and serves the simple needs of calculating start / end time here